UI/EnemyDetailsPanel.gd

```diff
@@ -91,125 +91,118 @@ func _on_enemy_changed():
-		var index = $VBoxContainer/GridContainer/IsBossBGM.pressed
-		SelectedListManager.apply_values_to_selected_type("is_boss_bgm", int(index))
+		SelectedListManager.apply_values_to_selected_type("is_boss_bgm", int(enemy.is_boss_bgm))
+		
 func _on_IsManualSet_pressed():
 	enemy.is_manual_set = $VBoxContainer/GridContainer/IsManualSet.pressed
 	if supress_event != true:
-		var index = $VBoxContainer/GridContainer/IsManualSet.pressed
-		SelectedListManager.apply_values_to_selected_type("is_manual_set", int(index))
+		SelectedListManager.apply_values_to_selected_type("is_manual_set", int(enemy.is_manual_set))
 
 func _on_IsAreaBoss_pressed():
 	enemy.is_area_boss = $VBoxContainer/GridContainer/IsAreaBoss.pressed
 	if supress_event != true:
-		var index = $VBoxContainer/GridContainer/IsAreaBoss.pressed
-		SelectedListManager.apply_values_to_selected_type("is_area_boss", int(index))
+		SelectedListManager.apply_values_to_selected_type("is_area_boss", int(enemy.is_area_boss))
 
 func _on_IsBloodEnemy_pressed():
 	enemy.is_blood_enemy = $VBoxContainer/GridContainer/BloodOrbsContainer/IsBloodEnemy.pressed
 	if supress_event != true:
-		var index = $VBoxContainer/GridContainer/BloodOrbsContainer/IsBloodEnemy.pressed
-		SelectedListManager.apply_values_to_selected_type("is_blood_enemy", int(index))
+		SelectedListManager.apply_values_to_selected_type("is_blood_enemy", int(enemy.is_blood_enemy))
 
 func _on_IsHighOrbEnemy_pressed():
 	enemy.is_highorb_enemy = $VBoxContainer/GridContainer/HighOrbsContainer/IsHighOrbEnemy.pressed
 	if supress_event != true:
-		var index = $VBoxContainer/GridContainer/HighOrbsContainer/IsHighOrbEnemy.pressed
-		SelectedListManager.apply_values_to_selected_type("is_highorb_enemy", int(index))
+		SelectedListManager.apply_values_to_selected_type("is_highorb_enemy", int(enemy.is_highorb_enemy))
 
 func _on_BloodOrbsSpinBox_value_changed(value):
 	enemy.blood_orbs = int(value)
 	if supress_event != true:
-		SelectedListManager.apply_values_to_selected_type("blood_orbs", int(value))
+		SelectedListManager.apply_values_to_selected_type("blood_orbs", int(enemy.blood_orbs))
 
 func _on_HighOrbsSpinBox_value_changed(value):
 	enemy.high_orbs = int(value)
 	if supress_event != true:
-		SelectedListManager.apply_values_to_selected_type("high_orbs", int(value))
+		SelectedListManager.apply_values_to_selected_type("high_orbs", int(enemy.high_orbs))
 
 func _on_ExpSpinBox_value_changed(value):
 	enemy.experience = int(value)
 	if supress_event != true:
-		SelectedListManager.apply_values_to_selected_type("experience", int(value))
+		SelectedListManager.apply_values_to_selected_type("experience", int(enemy.experience))
 
 func _on_DropsContainer_drops_table_selected(drops_table):
 	enemy.drops_table = drops_table
 	if supress_event != true:
-		var index = drops_table.id
-		SelectedListManager.apply_values_to_selected_type("drops_table", int(index))
+		SelectedListManager.apply_dropstable_to_selected("drops_table", Object(enemy.drops_table))
```

Comment:
I don't think you have to wrap the values in int() Object() etc. They should already by in the needed type